redash/query_runner/drill.py

```diff
@@ -0,0 +1,179 @@
+    'FLOAT': TYPE_FLOAT,
+    'REAL': TYPE_FLOAT,
+    'BOOLEAN': TYPE_BOOLEAN,
+    'bool': TYPE_BOOLEAN,
+    'TIMESTAMP': TYPE_DATETIME,
+    'TIME': TYPE_DATETIME,
+    'DATE': TYPE_DATETIME,
+    'CHAR': TYPE_STRING,
+    'STRING': TYPE_STRING,
+    'str': TYPE_STRING,
+    'VARCHAR': TYPE_STRING,
+    'CHARACTER': TYPE_STRING,
+    'CHARACTER VARYING': TYPE_STRING,
+    # Unsure about the interval type
+    'INTERVAL': TYPE_DATETIME,
+}
+
+
+class Drill(BaseQueryRunner):
+    @classmethod
+    def configuration_schema(cls):
+        return {
+            "type": "object",
+            "properties": {
+                "ConnectionType": {
+                    "type": "string"
+                },
+                "AuthenticationType": {
+                    "type": "string"
+                },
+                "host": {
+                    "type": "string"
+                },
+                "port": {
+                    "type": "number"
+                },
+                "ZKQuorum": {
+                    "type": "string"
+                },
+                "ZKClusterID": {
+                    "type": "string"
+                },
+            },
+            "required": ["ConnectionType", "AuthenticationType"]
+        }
+
+    @classmethod
+    def annotate_query(cls):
+        return False
+
+    @classmethod
+    def type(cls):
+        return "drill"
+
+    def __init__(self, configuration_json):
+        super(Drill, self).__init__(configuration_json)
+
+    def get_schema(self, get_stats=False):
+        try:
+            schemas_query = "select SCHEMA_NAME, TYPE from INFORMATION_SCHEMA.`SCHEMATA`"
+            tables_query = "show tables in %s"
+            files_query = "show files in %s"
+
+            columns_query = "select * from %s.`%s` limit 1"
+            schema = {}
+            for schema_name, schema_type in map(lambda a: (a['SCHEMA_NAME'], a['TYPE']), self._run_query_internal(schemas_query)):
+                # Bypassing drills default view, which should not be used if the data sources are properly defined
+                if 'default' in schema_name:
+                    continue
+                elif schema_type == 'file':
+                    query_to_run = files_query
+                elif schema_type == 'table':
+                    query_to_run = tables_query
+                else:
+                    continue
+
+                for table_name in map(lambda a: a['name'], self._run_query_internal(query_to_run % schema_name)):
+                    columns = map(lambda a: a['friendly_name'], self._run_query_internal_columns(columns_query % (schema_name, table_name)))
+
+                    if schema_name != 'INFORMATION_SCHEMA' and schema_name != 'sys':
+                        table_name = '{}.`{}`'.format(schema_name, table_name)
+
+                    schema[table_name] = {'name': table_name, 'columns': columns}
+        except Exception, e:
+            raise sys.exc_info()[1], None, sys.exc_info()[2]
```

Comment:
If you don't handle any exception and just re-raise it, why have the try/except at all?